git-gutter.el

```diff
@@ -966,13 +972,15 @@ start revision."
     (with-temp-file tmpfile
       (insert content))))
 
-(defsubst git-gutter:original-file-content (file)
+(defsubst git-gutter:original-file-content (file vcs)
   (with-temp-buffer
-    (when (zerop (process-file "git" nil t nil "show" (concat ":" file)))
-      (buffer-substring-no-properties (point-min) (point-max)))))
+    (cl-case vcs
+      (git (when (zerop (process-file "git" nil t nil "show" (concat ":" (file-name-nondirectory file))))
```

Comment:
I just noticed this PR after submitting #166.  I think your PR is great, but this moves a pre-existing bug: the second argument to `concat` needs to be a path relative to the `vc-git-root`.

Reply:
Thank you, Ross, you are right. I made my changes against `master`. I don't think any of these PRs will be merged any time soon, so I didn't include any of the other fixes. If at any point the maintainer decides to merge them, it will probably be the moment to merge all the fixes together. This way it is easier to see what each PR does and what it changes from the `master` reference.